Failure: `could not read Username for 'https://github.com': No such device or address`

I had a failure trying to release version 3.3.3 of the anupq package: it built the tarballs just fine but then failed to update the website.

See [this log](https://github.com/gap-packages/anupq/actions/runs/19673928076/job/56349740764#step:8:521) and  also [this log](https://github.com/gap-packages/anupq/actions/runs/19732395660/job/56536287089#step:8:522). 

```
...
create mode 100644 doc/times.css
 create mode 100644 doc/toggless.css
 create mode 100644 doc/toggless.js
fatal: could not read Username for 'https://github.com/': No such device or address
Error: Process completed with exit code 128.
```

This is coming from this script snippet:
```
git add -A
git commit --allow-empty -m "Update website for $PKGNAME $VERSION"
git push --set-upstream origin gh-pages
```


Now the error suggests some issue with authentication -- it is not using the `GITHUB_TOKEN` as it should perhaps. I could muck about with that, but fact is that it has worked for a ton of packages in the not so distant past, so I'd rather like to understand why it is failing now.


Since this is the very first time I've tried to use the action from the anupq repository, I am not sure if this is
- specific to the repository (i.e. some borked repo setting)
- a regression in this workflow (caused by a GitHub change)
- something else.

I will soon have to release another package and expect to see how it goes there, but I thought I should open this now in case others experience the same, and also in case e.g. @stertooy or @limakzi have any further insights into this.

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -21,17 +21,17 @@ jobs:
       # the order of the checkout actions is important because all contents of
       # the target folder of the checkout action is removed
       - name: "Clone the package"
-        uses: actions/checkout@v5
+        uses: actions/checkout@v6
         with:
           repository: ${{ matrix.package }}
 
       - name: "Check out this repository"
-        uses: actions/checkout@v5
+        uses: actions/checkout@v6
         with:
           path: .this-action/
 
       - name: "Setup GAP"
-        uses: gap-actions/setup-gap@v2
+        uses: gap-actions/setup-gap@v3
           
       - name: "Build the GitHub pages"
         uses: ./.this-action/

--- a/README.md
+++ b/README.md
@@ -52,8 +52,8 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v5
-      - uses: gap-actions/setup-gap@v2
+      - uses: actions/checkout@v6
+      - uses: gap-actions/setup-gap@v3
       - uses: gap-actions/update-gh-pages@v1
 ```
 
@@ -92,8 +92,8 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v5
-      - uses: gap-actions/setup-gap@v2
+      - uses: actions/checkout@v6
+      - uses: gap-actions/setup-gap@v3
       - uses: gap-actions/update-gh-pages@v1
         with:
           version: ${{ inputs.version }}

--- a/action.yml
+++ b/action.yml
@@ -41,7 +41,7 @@ runs:
         validate_boolean "${{ inputs.clean }}" clean
 
     # We explicitly do a fresh checkout so we don't rely on prevous workflow steps
-    # Until actions/checkout/issues/2318 is fixed, stick to v5
+    # Until https://github.com/actions/checkout/issues/2318 is fixed, stick to v5
     - name: "Checkout"
       uses: actions/checkout@v5
       with:

--- a/action.yml
+++ b/action.yml
@@ -40,8 +40,17 @@ runs:
         validate_boolean "${{ inputs.dry-run }}" dry-run
         validate_boolean "${{ inputs.clean }}" clean
 
+    # We explicitly do a fresh checkout so we don't rely on prevous workflow steps
+    # Until actions/checkout/issues/2318 is fixed, stick to v5
+    - name: "Checkout"
+      uses: actions/checkout@v5
+      with:
+        path: gh-pages-tmpdir
+        repository: ${{ inputs.repository }}
+
     - name: "Setup git"
       shell: bash
+      working-directory: gh-pages-tmpdir
       run: |
         git config --local user.email "github-actions[bot]@users.noreply.github.com"
         git config --local user.name "github-actions[bot]"
@@ -49,6 +58,7 @@ runs:
 
     - name: "Create gh-pages worktree"
       shell: bash
+      working-directory: gh-pages-tmpdir
       run: |
         if [[ -z $(git ls-remote --heads origin gh-pages) ]]; then
           echo "::warning::No existing 'gh-pages' branch was found, so it was created. You will still need to change the Pages settings of your repository."
@@ -62,7 +72,7 @@ runs:
     - name: "Setup GitHubPagesForGAP"
       shell: bash
       if: ${{ inputs.clean == 'true' }}
-      working-directory: gh-pages
+      working-directory: gh-pages-tmpdir/gh-pages
       run: |
         # Remove everything
         git rm -r --cached --ignore-unmatch *
@@ -106,7 +116,7 @@ runs:
 
     - name: "Copy files from the release"
       shell: bash
-      working-directory: gh-pages
+      working-directory: gh-pages-tmpdir/gh-pages
       run: |
         cp -f $TMP_DIR/PackageInfo.g .
         cp -f $TMP_DIR/$README .
@@ -127,7 +137,7 @@ runs:
 
     - name: "Copy extra files from the release"
       shell: bash 
-      working-directory: gh-pages
+      working-directory: gh-pages-tmpdir/gh-pages
       if: ${{ inputs.extra-files != '' }}
       run: |
         for PAIR in $(echo "${{ inputs.extra-files }}" | tr "," "\n"); do
@@ -154,7 +164,7 @@ runs:
 
     - name: "Update links in docs"
       shell: bash
-      working-directory: gh-pages
+      working-directory: gh-pages-tmpdir/gh-pages
       run: |
         if [ -d "doc" ] ; then
           cd doc
@@ -172,14 +182,14 @@ runs:
 
     - name: "Run update script"
       shell: bash
-      working-directory: gh-pages
+      working-directory: gh-pages-tmpdir/gh-pages
       run: |
         GAPROOT=${GAPROOT-$HOME/gap}
         $GAPROOT/gap update.g
 
     - name: "Update the gh-pages branch"
       shell: bash
-      working-directory: gh-pages
+      working-directory: gh-pages-tmpdir/gh-pages
       if: ${{ inputs.dry-run == 'false' }}
       run: |
         git add -A
@@ -190,7 +200,7 @@ runs:
       if: ${{ inputs.dry-run == 'true' }}
       uses: actions/jekyll-build-pages@v1
       with:
-        source: gh-pages
+        source: gh-pages-tmpdir/gh-pages
         destination: output
 
     - name: "Upload the release"

